cpp/log/cms_verifier_test.cc

```diff
@@ -0,0 +1,211 @@
+/* -*- indent-tabs-mode: nil -*- */
+#include <gflags/gflags.h>
+#include <glog/logging.h>
+#include <gtest/gtest.h>
+#include <openssl/err.h>
+#include <openssl/evp.h>
+#include <memory>
+#include <string>
+
+#include "log/cert.h"
+#include "log/cms_verifier.h"
+#include "log/ct_extensions.h"
+#include "util/status_test_util.h"
+#include "util/testing.h"
+#include "util/util.h"
+
+using cert_trans::Cert;
+using cert_trans::CmsVerifier;
+using std::string;
+using std::unique_ptr;
+using std::vector;
+using util::testing::StatusIs;
+
+// Self-signed
+static const char kCaCert[] = "ca-cert.pem";
+// Issued by ca-cert.pem
+static const char kLeafCert[] = "test-cert.pem";
+// Issued by ca-cert.pem
+static const char kIntermediateCert[] = "intermediate-cert.pem";
+
+// A DER file containing a CMS signed message wrapping data that is not
+// valid DER
+static const char kCmsSignedDataTest2[] = "cms_test2.der";
+// A DER file containing a CMS signed message wrapping a DER encoded
+// certificate for test case 3 (valid signature, same signer as cert)
+static const char kCmsSignedDataTest3[] = "cms_test3.der";
+// A DER file with a CMS signed message but not signed by the same
+// key as the certificate it contains in the payload
+static const char kCmsSignedDataTest4[] = "cms_test4.der";
+// A DER file with a CMS signed message with intermediate as signer and
+// issuer of the embedded cert
+static const char kCmsSignedDataTest5[] = "cms_test5.der";
+
+// Subject name we expect in our embedded certificate CMS tests
+static const char kCmsTestSubject[] = "CN=?.example.com, C=GB, ST=Wales, "
+    "L=Erw Wen, O=Certificate Transparency";
+
+namespace {
+
+class CmsVerifierTest : public ::testing::Test {
+ protected:
+  CmsVerifierTest()
+      : cert_dir_(FLAGS_test_srcdir + "/test/testdata"),
+        cert_dir_v2_(FLAGS_test_srcdir + "/test/testdata/v2/") {
+  }
+
+  string ca_pem_;
+  string intermediate_pem_;
+  string leaf_pem_;
+  const string cert_dir_;
+  const string cert_dir_v2_;
+  CmsVerifier verifier_;
+
+  void SetUp() {
+    CHECK(util::ReadTextFile(cert_dir_ + "/" + kLeafCert, &leaf_pem_))
+        << "Could not read test data from " << cert_dir_
+        << ". Wrong --test_srcdir?";
+    CHECK(util::ReadTextFile(cert_dir_ + "/" + kIntermediateCert,
+                             &intermediate_pem_));
+    CHECK(util::ReadTextFile(cert_dir_ + "/" + kCaCert, &ca_pem_));
+  }
+};
+
```

Comment:
Done. But I'm not sure it's a great idea as it can't be done automatically by formatters.